src/modules/roc_sndio/target_pulseaudio/roc_sndio/pulseaudio_backend.cpp

```diff
@@ -57,5 +57,19 @@ ISource* PulseaudioBackend::open_source(core::IAllocator&,
     return NULL;
 }
 
+void PulseaudioBackend::get_drivers(core::Array<DriverInfo>& arr) {
+    const char* format_name = "pulseaudio";
+    size_t n;
+    for (n = 0; n < arr.size(); n++) {
+        if (strcmp(format_name, arr[n].name) == 0) {
+            return;
+        }
+    }
+    if (arr.grow(arr.size() + 1)) {
+        DriverInfo newDriver(format_name);
+        arr.push_back(newDriver);
```

Comment:
I suggest to extract this code into a function, something like `add_driver_uniq(array, name)` and re-use it in PulseaudioBackend and SoxBackend.